srtcore/common.h

```diff
@@ -75,6 +75,98 @@ modified by
 #define SRT_ASSERT(cond)
 #endif
 
+#include <exception>
+
+// Class CUDTException exposed for C++ API.
+// This is actually useless, unless you'd use a DIRECT C++ API,
+// however there's no such API so far. The current C++ API for UDT/SRT
+// is predicted to NEVER LET ANY EXCEPTION out of implementation,
+// so it's useless to catch this exception anyway.
+
+class UDT_API CUDTException: public std::exception
+{
+public:
+
+   CUDTException(CodeMajor major = MJ_SUCCESS, CodeMinor minor = MN_NONE, int err = -1);
+   virtual ~CUDTException() ATR_NOTHROW {}
+
+      /// Get the description of the exception.
+      /// @return Text message for the exception description.
+
+   const char* getErrorMessage() const ATR_NOTHROW;
+
+   virtual const char* what() const ATR_NOTHROW ATR_OVERRIDE
+   {
+       return getErrorMessage();
+   }
+
+      /// Get the system errno for the exception.
+      /// @return errno.
+
+   int getErrorCode() const;
+
+      /// Get the system network errno for the exception.
+      /// @return errno.
+
+   int getErrno() const;
+      /// Clear the error code.
+
+   void clear();
+
+private:
+   CodeMajor m_iMajor;        // major exception categories
+   CodeMinor m_iMinor;		// for specific error reasons
+   int m_iErrno;		// errno returned by the system if there is any
+   mutable std::string m_strMsg; // text error message (cache)
```

Comment:
`m_iError` can be const.

Reply:
Unfortunately, not. See `CUDTException::clear()` in `common.cpp`.

Reply:
Look like clear() is only used in apps. And you are deleting that usage by this PR. See the following comments.

Reply:
Created #1032